app/services/katello/registration_manager.rb

```diff
@@ -152,7 +158,9 @@ def register_host(host, consumer_params, content_view_environments, activation_k
         new_host = host.new_record?
         unless new_host
           host.save!
-          unregister_host(host, :unregistering => true)
+          # Keep the kickstart repository ID so the host's Medium isn't unset
+          # Important for registering a host during provisioning
+          unregister_host(host, :reregistering => true, :keep_kickstart_repository => true)
```

Comment:
Shouldn't this be `unregistering`?
```suggestion
          unregister_host(host, :unregistering => true, :keep_kickstart_repository => true)
```